woocommerce/woocommerce-gutenberg-products-block
# Price Filter powered by Interactivity API

## What

Fixes #10813

This PR is the initial implementation of the new Price Filter block using Interactivity API:
- This block is an inner block of the `<CollectionFilter>` block.
- In the Editor, the collection data is retrieved using `useCollectionData` hook as the current Price Filter block does.
- On the frontend, the block retrieves collection data through a custom context passed down from `CollectionFilter` block.
- The block reuses `@woocommerce/base-components`/`FilterResetButton` component but not `PriceSlider` because the latter is too complicated for the editor preview and it's only used there.
- This is not the final version of the block, we want filter blocks to be extensible by design, we're investigating on using inner blocks for that purpose (GH discussion is incoming).

## Why
- Due to the limitation of rendering React component on the frontend, we can't really solve the performance issue of filter blocks (content flashing, time to interact).
- We found Interactivity API solve the performance nicely as the block now is SSR, there is no more content flashing and waiting for fetching collection data after page load.


## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. On a new page, add these blocks (the later is the inner block of the former):
  1.  Product Collection
  2. Collection Filters
  3. Collection Price Filter.
2. See the block preview renders correctly in the editor.
3. See no console warning or error.
4. Test the block on the front end.
5. See no content flashing.
6. See products update on the price filter change.
7. See the min price can't exceed the max.
8. Test Show Input Fields and Inline input settings, see editor preview reacts and frontend works as expected.


## Screenshots or screencast



https://github.com/woocommerce/woocommerce-blocks/assets/5423135/af343d25-2c33-4542-a709-652b80b683b6

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
